Tasks/AzureResourceGroupDeploymentV2/operations/DeploymentGroupExtensionHelper.ts

```diff
@@ -206,13 +206,21 @@ export class DeploymentGroupExtensionHelper {
             var publicSettings = {
                 VSTSAccountName: collectionUri,
                 TeamProject: teamProject,
-                MachineGroup: this.taskParameters.deploymentGroupName,
                 DeploymentGroup: this.taskParameters.deploymentGroupName,
                 AgentName: "",
                 Tags: tags
             };
             console.log("Public settings are:\n VSTSAccountName: %s\nTeamProject: %s\nDeploymentGroup: %s\nTags: %s\n", collectionUri, teamProject, this.taskParameters.deploymentGroupName, JSON.stringify(tags));
             var protectedSettings = { PATToken: this.taskParameters.tokenCredentials.getPatToken() };
+            if (this.taskParameters.runAgentServiceAsUser) {
+                if (vmOsType === "Windows") {
+                    publicSettings["WindowsLogonAccountName"] = this.taskParameters.agentServiceUserCredentials.getUserName();
```

Comment:
why do we have a different parameter for windows and linux

Reply:
Issue added in vm extension repo